### DISCUSSION TITLE
Handler::report(Throwable $e) must be compatible with...

### DISCUSSION THREAD

**Discussion by Voorsie** [laravel/framework#33098] [General] [answered]
So I have been reading all issues about this and trying for hours, but can't seem to fix it with the current information. The error I'm getting is;

```PHP Fatal error:  Declaration of Illuminate\Foundation\Exceptions\Handler::report(Throwable $e) must be compatible with Illuminate\Contracts\Debug\ExceptionHandler::report(Exception $e) in D:\home\site\wwwroot\vendor\laravel\framework\src\Illuminate\Foundation\Exceptions\Handler.php on line 38```

I have followed the upgrade guide and copied the default app/Exceptions/Handler.php file but I still get the error. Tried removing the vendor folder and running composer install, composer update, composer dump-autoload but nothing seems to fix the issue. Running on php 7.3.14

Am I looking in the wrong direction?

**Comment by plumpboy** [comment]
you just replace Throwable with Exception, this issue will be solved

**Reply by Voorsie** [reply]
Thanks for your reply @plumpboy, but like I mentioned I copied the default app/Exceptions/Handler.php file from the latest Laravel repo which do not use any Exeption.

Or do you mean in another file?

**Reply by mfn** [reply]
As @plumpboy said.

This is also documented in the upgrade guide, see https://laravel.com/docs/7.x/upgrade#symfony-5-related-upgrades

**Comment by denjaland** [comment, marked as answer]
Hi @Voorsie,

To make things more performant, Laravel caches quite some stuff in `/bootstrap/cache`.  I ran into the same issue before, and removing the /bootstrap/cache/compiled.php might actually resolve this.

Normally you can use `php artisan clear-compiled` to remove the compiled class file, but you're trapped in a catch 22 because in order to run the command, Laravel will use... the compiled cache file throwing the error you got.

So try removing the file manually, and you should be good to go.

**Reply by Voorsie** [reply]
Thanks @denjaland,

this was indeed the loop I was stuck in. Have tried clearing the cache through the php artisan functions but these would throw me the error.

Wasn't aware of this persistent cache file compiled.php but after checking that I found that the functions from the previous version was cached there and that was the problem of the error while running. 

After removing I was able to regenerate the cache without any problems and run my application!

**Comment by imanghafoori1** [comment]
This particular exception may be caught and handled by laravel...
in the handler, laravel can delete compiled.php file.

**Reply by denjaland** [reply]
I don't think you can catch all the exceptions that may be caused by this just to delete the file.  Also, this error doesn't always mean it is caused by a cached compiled.php file; it can be a genuine error...
